src/Maker/MakeCrud.php

```diff
@@ -0,0 +1,146 @@
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Sadicov Vladimir <sadikoff@gmail.com>
+ */
+final class MakeCrud extends AbstractMaker
+{
+    private $router;
+
+    private $locator;
+
+    public function __construct(RouterInterface $router, ContainerInterface $serviceLocator)
+    {
+        $this->router = $router;
+        $this->locator = $serviceLocator;
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:crud';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureCommand(Command $command, InputConfiguration $inputConfig)
+    {
+        $command
+            ->setDescription('Creates crud for Doctrine entity class')
+            ->addArgument('entity-class', InputArgument::OPTIONAL, sprintf('The class name of the entity to create crud (e.g. <fg=yellow>%s</>)', Str::asClassName(Str::getRandomTerm())))
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeCrud.txt'))
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+        $dependencies->addClassDependency(
+            Route::class,
+            'annotations'
+        );
+
+        $dependencies->addClassDependency(
+            TwigBundle::class,
+            'twig-bundle'
+        );
+
+        $dependencies->addClassDependency(
+            EntityManager::class,
+            'orm-pack'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParameters(InputInterface $input): array
+    {
+        $entityClassName = Str::asClassName($input->getArgument('entity-class'));
+        Validator::validateClassName($entityClassName);
+        $controllerClassName = Str::asClassName($entityClassName, 'Controller');
+        Validator::validateClassName($controllerClassName);
+        $formClassName = Str::asClassName($entityClassName, 'Type');
+        Validator::validateClassName($formClassName);
+
+        $metadata = $this->getEntityMetadata($entityClassName);
+
+        return array(
+            'controller_class_name' => $controllerClassName,
+            'entity_var_plural' => lcfirst(Inflector::pluralize($entityClassName)),
+            'entity_var_singular' => lcfirst(Inflector::singularize($entityClassName)),
+            'entity_class_name' => $entityClassName,
+            'entity_identifier' => $metadata->identifier[0],
+            'entity_fields' => $metadata->fieldMappings,
```

Comment:
So this is a great start. And actually, we need to improve `make:form` to be smarter and do stuff like this - see #114.

But, we need to be a bit smarter, and fortunately, this work was done already in SensioGeneratorBundle :). See: https://github.com/sensiolabs/SensioGeneratorBundle/blob/master/Generator/DoctrineFormGenerator.php#L99

The eliminate the identifier fields more cleanly and avoid adding OneToMany relationships - which will definitely not work :). I think we could get even crazier, but let's just match what was done before.

And, we could (in this same PR for convenience) update `make:form` to use this new awesomeness as well :). But, we can do it later if you'd like.

Reply:
I can improve make:form in this PR, but I need an advise. How smarter this command must be?
For example:
We create 2 ways for command, 
[1] there is Entity with enetered name
[2] there is **NO** Entity

If we have entity, than we generate complete form with all fields, otherwise we generate only stub class.

Is this will be enough? or we need something more?

Reply:
That’s it :).

Except, I also think we need to change the initial question:

> Do you want to generate a form for an entity?

If yes, ask them for the entity class (and build the form class name from that). If no, ask them the form class.

Reply:
Ok, that's clear.

And one question about using this command from crud generation. I think we need at first generate form, than crud. That's why I think we need to add something like beforeGenerate method. What do you think about it?

Reply:
Actually, in one of the PR’s, we completely simplify into just a generate() method :). But I get your point: we’re generating a form, then a crud. But this should be invisible to the user: they’re just thinking: in generating a crud based on an entity. So we shouldn’t try to call the make:form command from make:crud. But we should isolate any generation logic that needs to be reused into a service and then call it from both commands.

Reply:
If you don't mind I took DoctrineMetadataFactory from your PR =) and pushed improved make:form command

Reply:
It's no problem - good idea if you needed it :)